### ISSUE TITLE
Api `spec_kwargs` param not working while `app` is None

### ISSUE THREAD

**Issue by jan-golda** [marshmallow-code/flask-smorest#103]
I tried to use `spec_kwargs` param, but it had no effect what so ever, so I looked in to the code and found this:

This works as expected:
```python
api = Api(app, spec_kwargs={...})
```
This also works:
```python
api = Api()
api.init_app(app, spec_kwargs={...})
```
But this does not:
```python
api = Api(spec_kwargs={...})
api.init_app(app)
```

That's because `spec_kwargs` from `__init__` is not passed to `init_app` (if executed separately):
https://github.com/marshmallow-code/flask-smorest/blob/720df5efebbf9316cf6d63e5b7afc315db2040b4/flask_smorest/__init__.py#L40-L60

That's a bit unexpected, so I think this should be mentioned in documentation or fixed :smile:

**Comment by lafrech** [comment]
Thanks for reporting.

Fix incoming: #104.

**Comment by lafrech** [comment]
What would you expect if some kwargs are passed at init, then different kwargs (even empty dict) are passed at init_app?

Merge? Override?

**Comment by jan-golda** [comment]
Good question, but probably override, it's simpler.
As long as this will be stated in docs either is fine.

**Comment by lafrech** [comment]
I implemented "override" already, but on second thought, I think "update" would be simpler, less surprising.

I'll update my PR.

**Comment by jan-golda** [comment]
Yeah, that's true, "update" is less surprising.
So therefore probably that's the better choice

**Comment by lafrech** [comment]
Fix released in 0.18.1.
